src/charm.py

```diff
@@ -225,6 +246,13 @@ def _on_config_changed(self, event: EventBase):
             event.defer()
             return
         self._on_bessd_pebble_ready(event)
+        if self._restart_pod:
+            try:
+                self._kubernetes_multus.delete_pod()
```

Comment:
I don't understand much the order on the execution flow. But why basing the decision of restarting the pod over a flag set on _network_attachment_definitions_from_config method instead of detecting the config change here are deleting the pod right away?

Reply:
KubernetesMultusCharmLib library run _configure_multus method which deletes the old NAD definitions and creates new NAD definitions with latest config settings when config_changed event is fired. In config_changed method of charm if we try to detect if pod restart is required or not, it returns False. So this check needs to be done before config_changed run and restart needs to be done after new NADs are created to make the new NADs take effect.

```
self.framework.observe(charm.on.config_changed, self._configure_multus)
```